include/neml2/models/crystallography/CrystalGeometry.h

```diff
@@ -0,0 +1,168 @@
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#pragma once
+
+#include "neml2/models/StaticModel.h"
+#include "neml2/tensors/BatchTensorBase.h"
+
+namespace neml2
+{
+class Vec;
+class R2;
+class SR2;
+class WR2;
+namespace crystallography
+{
+class CrystalClass;
+class MillerIndex;
+
+/// @brief Defines the geometry of a crystal system
+/// This includes a basic definition of the crystal lattice,
+/// via Bravais vectors and a CrystalClass object defining the
+/// crystal symmetry as well as the definition of the geometry
+/// of each slip system.
+class CrystalGeometry : public StaticModel
+{
+public:
+  /// Setup from parameter set
+  CrystalGeometry(const OptionSet & options);
+
+  /// Alternate constructor not relying on options
+  CrystalGeometry(const OptionSet & options,
+                  const CrystalClass & cclass,
+                  const Vec & lattice_vectors);
+
+  /// Input options
+  static OptionSet expected_options();
+
+  /// accessor for the first lattice vector
+  Vec a1() const;
+  /// accessor for the second lattice vector
+  Vec a2() const;
+  /// accessor for the third lattice vector
+  Vec a3() const;
+
+  /// accessor for the first reciprocal lattice vector
+  Vec b1() const;
+  /// accessor for the second reciprocal lattice vector
+  Vec b2() const;
+  /// accessor for the third reciprocal lattice vector
+  Vec b3() const;
+
+  /// Total number of slip systems
+  TorchSize nslip() const;
+  /// Number of slip groups
+  TorchSize nslip_groups() const;
+  /// Number of slip systems in a given group
+  TorchSize nslip_in_group(TorchSize i) const;
+
+  /// Accessor for the slip directions
+  const Vec & cartesian_slip_directions() const { return _cartesian_slip_directions; };
+  /// Accessor for the slip planes
+  const Vec & cartesian_slip_planes() const { return _cartesian_slip_planes; };
+  /// Accessor for the burgers vector
+  const Scalar & burgers() const { return _burgers; };
+
+  /// Accessor for the full Schmid tensors
+  const R2 & A() const { return _A; };
+  /// Accessor for the symmetric Schmid tensors
+  const SR2 & M() const { return _M; };
+  /// Accessor for the skew-symmetric Schmid tensors
+  const WR2 & W() const { return _W; };
+
+  /// Accessor for the crystal class
+  const CrystalClass & crystal_class() const { return _class; };
```

Comment:
I don't think we need accessors now that we have `Model::get_buffer`.

Reply:
I still like to have them, at least for commonly-used stuff.